codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2GoAwayFrame.java

```diff
@@ -0,0 +1,172 @@
+        this(error.code(), content);
+    }
+
+    /**
+     * Construct a new GOAWAY message.
+     *
+     * @param error reason for the go away
+     * @param content non-{@code null} debug data
+     */
+    public DefaultHttp2GoAwayFrame(long errorCode, ByteBuf content) {
+        this.errorCode = errorCode;
+        this.content = checkNotNull(content, "content");
+    }
+
+    @Override
+    public long errorCode() {
+        return errorCode;
+    }
+
+    @Override
+    public int extraStreamIds() {
+        return extraStreamIds;
+    }
+
+    @Override
+    public DefaultHttp2GoAwayFrame setExtraStreamIds(int extraStreamIds) {
+        if (extraStreamIds < 0) {
+            throw new IllegalArgumentException("extraStreamIds must be non-negative");
+        }
+        this.extraStreamIds = extraStreamIds;
+        return this;
+    }
+
+    @Override
+    public ByteBuf content() {
+        if (content.refCnt() <= 0) {
+            throw new IllegalReferenceCountException(content.refCnt());
+        }
+        return content;
+    }
+
+    @Override
+    public DefaultHttp2GoAwayFrame copy() {
+        return new DefaultHttp2GoAwayFrame(errorCode, content().copy()).setExtraStreamIds(extraStreamIds);
+    }
+
+    @Override
+    public DefaultHttp2GoAwayFrame duplicate() {
+        return new DefaultHttp2GoAwayFrame(errorCode, content().duplicate()).setExtraStreamIds(extraStreamIds);
+    }
+
+    @Override
+    public int refCnt() {
+        return content.refCnt();
+    }
+
+    @Override
+    public boolean release() {
+        return content.release();
+    }
+
+    @Override
+    public boolean release(int decrement) {
+        return content.release(decrement);
+    }
+
+    @Override
+    public DefaultHttp2GoAwayFrame retain() {
+        content.retain();
+        return this;
+    }
+
+    @Override
+    public DefaultHttp2GoAwayFrame retain(int increment) {
+        content.retain(increment);
+        return this;
+    }
+
+    @Override
+    public String toString() {
+        return "DefaultHttp2GoAwayFrame(errorCode=" + errorCode + ",content=" + content
+            + ",extraStreamIds=" + extraStreamIds + ")";
+    }
+
+    @Override
+    public DefaultHttp2GoAwayFrame touch() {
+        content.touch();
+        return this;
+    }
+
+    @Override
+    public DefaultHttp2GoAwayFrame touch(Object hint) {
+        content.touch(hint);
+        return this;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (!(o instanceof DefaultHttp2GoAwayFrame)) {
+            return false;
+        }
+        DefaultHttp2GoAwayFrame other = (DefaultHttp2GoAwayFrame) o;
+        return super.equals(o) && errorCode == other.errorCode && content.equals(other.content)
+            && extraStreamIds == other.extraStreamIds;
+    }
+
+    @Override
+    public int hashCode() {
+        int hash = 237395317;
+        hash = hash * 31 + (int) (errorCode ^ (errorCode >>> 32));
```

Comment:
Also only available in Java 8. If we don't need Java 7 compat, then I can change.